src/main/dynamo/Beyonce.ts

```diff
@@ -144,40 +144,30 @@ export class Beyonce {
     keys: T[]
     consistentRead?: boolean
   }): Promise<GroupedModels<ExtractKeyType<T>>> {
-    const useConsistentRead = params.consistentRead !== undefined ? params.consistentRead : this.consistentReads
-
-    const { Responses: responses, UnprocessedKeys: unprocessedKeys } = await this.client
-      .batchGet({
-        RequestItems: {
-          [this.table.tableName]: {
-            ConsistentRead: useConsistentRead,
-            Keys: params.keys.map(({ partitionKey, sortKey }) => ({
-              [this.table.partitionKeyName]: partitionKey,
-              [this.table.sortKeyName]: sortKey
-            }))
-          }
-        }
-      })
-      .promise()
+    const { keys, consistentRead } = params
 
-    if (unprocessedKeys !== undefined && Object.keys(unprocessedKeys).length > 0) {
-      throw new Error(`Some keys didn't process: ${JSON.stringify(unprocessedKeys)}`)
+    if (keys.length === 0) {
+      groupModelsByType<ExtractKeyType<T>>([], [])
     }
 
-    const modelTags = params.keys.map((_) => _.modelTag)
-
-    if (responses !== undefined) {
-      const items = responses[this.table.tableName]
-      const jsonItemPromises = items.map(async (item) => {
-        const maybeDecryptedItem = await decryptOrPassThroughItem(this.jayz, item)
-        return maybeDecryptedItem as ExtractKeyType<T>
-      })
-
-      const jsonItems = await Promise.all(jsonItemPromises)
-      return groupModelsByType(jsonItems, modelTags)
-    } else {
-      return groupModelsByType<ExtractKeyType<T>>([], modelTags)
+    const modelTags = keys.map((_) => _.modelTag)
+    const requestsInFlight: Promise<ExtractKeyType<T>[]>[] = []
+
+    // Chunk keys into N concurrent requests
+    // as DynamoDB has a max of 100 keys per batchGet requests
+    const batchSize = 100
+    let batchStartIndex = 0
+    let batchEndIndex = batchSize
+    while (batchStartIndex < keys.length) {
+      const batchedKeys = keys.slice(batchStartIndex, batchEndIndex)
```

Comment:
The start index passed to slice [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) is included, but the end index is not. 

So `[1,2,3].slice(0, 2)` => `[1,2]`. 

And you can specify an end index larger than the array without it complaining, so: 
`[1,2,3].slice(0, 10000)` => `[1,2,3]`

More concretely in this loop: 

```TypeScript
// 1st iteration
batchStartIndex == 0
batchEndIndex == 100
keys.slice(0, 100) =>  1st 100 elements

// 2nd iteration
batchStartIndex = 100
batchEndIndex = 200
keys.slice(100, 200) => 2nd 100 elements

```